modules/vstudio/vs2005_dotnetbase.lua

```diff
@@ -826,36 +826,35 @@
             ["WindowsDesktop"] = "Microsoft.NET.Sdk.WindowsDesktop",
             ["MSTest"] = "MSTest.Sdk",
         }
+		local parts = nil
+		if cfg.dotnetsdk then
+			parts = cfg.dotnetsdk:explode("/", true, 1)
+		end
 
-		if cfg.flags.WPF then
-			return map["WindowsDesktop"]
-        end
+		local sdk = (parts and #parts > 0 and parts[1]) or cfg.dotnetsdk
+		if not parts or #parts < 2 then
+			-- prioritize WPF flag for compatibility
+			if cfg.flags.WPF then
+				return map["WindowsDesktop"]
+			end
```

Comment:
This probably won't work as expected - it doesn't trigger if a user doesn't specify a `dotnetsdk` value, which is the current behaviour.

Deprecating the flag would eliminate this need. In this block:
https://github.com/premake/premake-core/blob/6f0cb4a7ea2aa5d7d1b6024c7d15b82f039efb76/src/_premake_init.lua#L1112-L1129

You'll want to add something like:
```lua
api.deprecateValue("flags", "WPF", 'Use `dotnetsdk "WindowsDesktop"` instead.',
function(value)
  dotnetsdk "WindowsDesktop"
end,
function(value)
  dotnetsdk "Default"
end)
```